dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx

```diff
@@ -387,4 +442,35 @@ export class AppRepoForm extends React.Component<IAppRepoFormProps, IAppRepoForm
   private handleSyncJobPodTemplateChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
     this.setState({ syncJobPodTemplate: e.target.value, validated: undefined });
   };
+
+  private togglePullSecret = (imagePullSecret: string) => {
+    return () => {
+      const { selectedImagePullSecrets } = this.state;
+      this.setState({
+        selectedImagePullSecrets: {
+          ...selectedImagePullSecrets,
+          [imagePullSecret]: !selectedImagePullSecrets[imagePullSecret],
+        },
+      });
+    };
+  };
+
+  // Select the pull secrets based on the current status and if they are already
+  // selected in the existing repo info
+  private parseSecrets = (secrets: ISecret[], repo?: IAppRepository) => {
+    const selectedImagePullSecrets = this.state.selectedImagePullSecrets;
+    secrets.forEach(secret => {
+      let selected = false;
+      // If it has been already selected
+      if (selectedImagePullSecrets[secret.metadata.name]) {
+        selected = true;
+      }
+      // Or if it's already selected in the existing repo
+      if (repo?.spec?.dockerRegistrySecrets?.some(s => s === secret.metadata.name)) {
+        selected = true;
+      }
```

Comment:
Here I guess: if the secret on the app repo *isn't* available, we have an issue (or they should be explicitly removed by the user from the app repo).

Reply:
yes, missing secrets won't be shown.